templates/ros2_cpp_pkg/{{ package_name }}/src/{{ node_name }}.cpp.jinja

```diff
@@ -219,25 +193,27 @@ void {{ node_class_name }}::setup() {
 {% if has_subscriber %}
 
 
-/**
- * @brief Processes messages received by a subscriber
- *
- * @param msg message
- */
-void {{ node_class_name }}::topicCallback(const std_msgs::msg::Int32& msg) {
+void {{ node_class_name }}::topicCallback(const std_msgs::msg::Int32::ConstSharedPtr& msg) {
 
-  RCLCPP_INFO(this->get_logger(), "Message received: '%d'", msg.data);
+  RCLCPP_INFO(this->get_logger(), "Message received: '%d'", msg->data);
+
+  // publish message
+{% if is_component %}
+  std_msgs::msg::Int32::UniquePtr out_msg = std::make_unique<std_msgs::msg::Int32>();
+  out_msg->data = msg->data;
+  publisher_->publish(std::move(out_msg));
+  RCLCPP_INFO(this->get_logger(), "Message published: '%d'", out_msg->data);
+{% else %}
+  std_msgs::msg::Int32 out_msg;
```

Comment:
Why don't we also use a unique pointer here?

Reply:
Easier syntax, doesn't matter for non-components.